brandon/server.js

```diff
@@ -0,0 +1,217 @@
+const fs = require('fs');
+const express = require('express');
+const pg = require('pg');
+
+// REVIEW: Require in body-parser for post requests in our server. If you want to know more about what this does, read the docs!
+const bodyParser = require('body-parser');
+const PORT = process.env.PORT || 3000;
+const app = express();
+
+
+// DONE: Complete the connection string for the url that will connect to your local postgres database
+// Windows and Linux users; You should have retained the user/pw from the pre-work for this course.
+// Your url may require that it's composed of additional information including user and password
+// const conString = 'postgres://USER:PASSWORD@HOST:PORT/DBNAME';
+const conString = 'postgres://localhost:5432/lab-08';
+
+// DONE: Our pg module has a Client constructor that accepts one argument: the conString we just defined.
+//       This is how it knows the URL and, for Windows and Linux users, our username and password for our
+//       database when client.connect is called on line 26. Thus, we need to pass our conString into our
+//       pg.Client() call.
+const client = new pg.Client(conString);
+
+// REVIEW: Use the client object to connect to our DB.
+client.connect();
+
+
+// REVIEW: Install the middleware plugins so that our app is aware and can use the body-parser module
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: true}));
+app.use(express.static('./public'));
+
+
+// REVIEW: Routes for requesting HTML resources
+app.get('/new', function(request, response) {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // Put your response here...
+
+  /* This is using numbers 2 and 5 by requesting localhost:3000/new and the server is responding with sending the user to new.html.  This method is interacting with the Article.fetchAll method on line 45 in article.js.  This is the Read part of CRUD.*/
+
+  response.sendFile('new.html', {root: './public'});
+});
+
+
+// REVIEW: Routes for making API calls to use CRUD Operations on our database
+app.get('/articles', function(request, response) {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // Put your response here...
+
+  /* This is using steps 3 and 4 by selecting everything from articles and then running the funciton which gets the response to send the data from our table.  This is also interacting with the Article.fetchAll method in article.js It is using the read part of CRUD */
```

Comment:
It also uses 2 and 5. The 2 is a request from the user to get all articles, and the 5 is the system's response to the user.